### PR TITLE
Add handling of copy, cut, paste, select all

### PR DESCRIPTION
**Pull request by JohannesDeml**
Add support for common shortcuts:
- Copy selected text
- Cut selected text
- Paste Text from Clipboard
- Select complete command input

Missing Support:
- Undo & Redo (Sadly not that easy)

Still, I think this makes the UX a lot better. I was missing Copy&Paste a lot.

### CODE REVIEW

**Comment by marijnz** [comment]
Hey! Thanks for contributing more.

For me, copy, cut, paste all already work. Only selecting the complete command input doesn't yet. I'm on Mac, are you on Windows and it maybe that it?

**Comment by JohannesDeml** [comment]
Hmmmm, very interesting. Yes, I'm on windows, that might be the problem. That's quite strange that the logic only needs to be applied for windows!
Is there unexpected behavior when integrating my PR on mac? I had the problem with multiple unity versions on different machines (all windows). Might be interesting though, if others do expect that problem as well and how unity handles the commands on linux.

Anyways, I added another small fix when the cursor is at the begging of the command and there is a selection you want to delete. Backspace was not working for that case.

Hm, so how should we procede here? Do you have the possibility to test the functionality of windows? Should I add platform dependant compilation?

**Comment by JohannesDeml** [comment]
Dang it, that's kind of what I feared when you said it is already working for you.
Yep, I guess TextEditor would be the first place to look at (https://github.com/Unity-Technologies/UnityCsReference/blob/11bcfd801fccd2a52b09bb6fd636c1ddcc9f1705/Modules/IMGUI/TextEditor.cs). Sadly I don't have the time either today. If there is anyone who wants to take it on from here, I would be happy to pass it on, otherwise I will get to the root as soon as I find the time :)

Thanks for rechecking everything with macOS, I thought the problem would be the same there. If we can only solve it with conditional compilation I will encapsulate the scripts.
The logic that makes sense on all platforms is select all, right?

File: Assets/UnityShell/Editor/Scripts/UnityShellEditorWindow.cs [hunk 1]
```diff
@@ -143,21 +143,68 @@ private void OnInspectorUpdate()
...
+            HandleTextCommands();
 			autocompleteBox.HandleEvents();
 			HandleHistory();
 			DoAutoComplete();
 			HandleRequests();
 			DrawAll();
 		}
 
-		private void HandleHistory()
+        private void HandleTextCommands()
```
[the remark this answers is not in the document]
Comment (JohannesDeml) [reply Assets/UnityShell/Editor/Scripts/UnityShellEditorWindow.cs]: Hey Wolfreak99,
thanks for the review. Ugh, yuck, you're completely right. seems like an ugly space tabs mixing. My bad, I will fix it as soon as we know how to proceed with the final integration. 👍

**Comment by marijnz** [comment]
If this is still a thing (of course it is..) I guess we can go with the separate code for Windows way?

**Comment by JohannesDeml** [comment]
Sorry for just leaving this PR here without doing anything. I'm actually on vacation right now, so I won't do anything within the next days (probably).
Yep, it is still a thing on windows and linux. I think the best way is to close this PR and I will do a new clean implementation with separate code for windows and linux, as you suggested (and in a new class to avoid cluttering the logic).

I did look a bit through the decompiled code, but did find anything that explains the behavior (I just took a glimpse), so I think we will just go with compile dependant code without knowing why it is necessary ;)

**Pull request closed without merging**